eng/scripts/Language-Settings.ps1

```diff
@@ -172,58 +172,97 @@ function Get-javascript-GithubIoDocIndex()
   GenerateDocfxTocContent -tocContent $tocContent -lang "JavaScript"
 }
 
-# Updates a js CI configuration json.
-# For "latest", we simply set a target package name
-# For "preview", we add @next to the target package name
-function Update-javascript-CIConfig($pkgs, $ciRepo, $locationInDocRepo, $monikerId = $null)
-{
-  $pkgJsonLoc = (Join-Path -Path $ciRepo -ChildPath $locationInDocRepo)
-
-  if (-not (Test-Path $pkgJsonLoc))
-  {
-    Write-Error "Unable to locate package json at location $pkgJsonLoc, exiting."
-    exit(1)
+# "@azure/package-name@1.2.3" -> "@azure/package-name"
+function Get-PackageNameFromDocsMsConfig($DocsConfigName) { 
+  if ($DocsConfigName -match '^(@?.+?)(@(.*))?$') { 
+    return $Matches[1]
   }
 
-  $allJson = Get-Content $pkgJsonLoc | ConvertFrom-Json
+  throw "Cannot find package $DocsConfigName"
+}
 
-  $visibleInCI = @{}
+function Update-javascript-DocsMsPackages($DocsRepoLocation)
+{
+  $publishedPackages = Get-CSVMetadata `
+  | Where-Object { $_.New -eq 'true' -and $_.Hide -ne 'true' }
+
+  # Update "Latest" packages
+  $latestConfigPath = Join-Path $DocRepoLocation '/ci-configs/packages-latest.json' -Resolve
+  Write-Host "Updating latest configuration: $latestConfigPath"
+  $latestPackages = Get-Content $latestConfigPath -Raw | ConvertFrom-Json
+
+  $gaPackages = $publishedPackages | Where-Object { $_.VersionGA.trim() }
+  foreach ($package in $gaPackages) { 
+    if ($latestPackages.npm_package_sources.name -contains $package.Package) { 
+      # Onboarded packages already contains this package, skip
+      Write-Host "Keeping onboarded package $($package.Package)"
+      continue
+    }
 
-  for ($i = 0; $i -lt $allJson.npm_package_sources.Length; $i++)
-  {
-    $pkgDef = $allJson.npm_package_sources[$i]
-    $accessor = ($pkgDef.name).Replace("`@next", "")
-    $visibleInCI[$accessor] = $i
+    Write-Host "Add GA package: $($package.Package)"
+    $latestPackages.npm_package_sources += @{ name = $package.Package }
   }
 
-  foreach ($releasingPkg in $pkgs)
-  {
-    $name = $releasingPkg.PackageId
-
-    if ($releasingPkg.IsPrerelease)
-    {
-      $name += "`@next"
+  $latestPackages | ConvertTo-Json -Depth 100 | Set-Content $latestConfigPath
+
+  # Update "preview" packages
+  $previewConfigPath = Join-Path $DocRepoLocation '/ci-configs/packages-preview.json' -Resolve
+  Write-Host "Updating preview configuration: $previewConfigPath"
+  $previewPackages = Get-Content $previewConfigPath -Raw | ConvertFrom-Json
+
+  $outputPackages = @()
+  foreach ($package in $previewPackages.npm_package_sources) {
+    # Do not filter by GA/Preview status because we want differentiate between
+    # tracked and non-tracked packages
+    $matchingPublishedPackages = $publishedPackages `
+    | Where-Object { $_.Package -eq (Get-PackageNameFromDocsMsConfig $package.name) }
```

Comment:
```suggestion
    $matchingPublishedPackages = $publishedPackages.Where({ $_.Package -eq (Get-PackageNameFromDocsMsConfig $package.name) })
```

We should consider using Where function instead of Where-Object pipeline because the Where function always guarantees you get back a list. Have a look at this [teams thread](https://teams.microsoft.com/l/message/19:59dbfadafb5e41c4890e2cd3d74cc7ba@thread.skype/1607623901212?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1607623901212&teamName=Azure%20SDK&channelName=Engineering%20System%20%F0%9F%9B%A0%EF%B8%8F&createdTime=1607623901212) if you want more information.